packages/features/bookings/tests/handleNewBooking.test.ts

```diff
@@ -0,0 +1,131 @@
+import { PrismaClient } from "@prisma/client";
+import type { Request, Response } from "express";
+import type { NextApiRequest, NextApiResponse } from "next";
+import { createMocks } from "node-mocks-http";
+import { test } from "vitest";
+
+// import { default as handleNewBooking } from "@calcom/features/bookings/lib/handleNewBooking";
+// import prisma from "@calcom/prisma";
+import { getBooker, getDate } from "@calcom/web/test/utils/bookingScenario";
+
+type CustomNextApiRequest = NextApiRequest & Request;
+type CustomNextApiResponse = NextApiResponse & Response;
+
+const prisma = new PrismaClient();
+
+export function withPrisma(handler: any) {
+  return async (req: any, res: any) => {
+    if (!("prisma" in req)) {
+      req.prisma = prisma;
+    }
+    return handler(req, res);
+  };
+}
+
+test("handleNewBooking", async () => {
+  const handleNewBooking = (await import("@calcom/features/bookings/lib/handleNewBooking")).default;
+  const booker = getBooker({
+    email: "booker@example.com",
+    name: "Booker",
+  });
+
+  //   const organizer = getOrganizer({
+  //     name: "Organizer",
+  //     email: "organizer@example.com",
+  //     id: 101,
+  //     schedules: [TestData.schedules.IstWorkHours],
+  //     credentials: [getGoogleCalendarCredential()],
+  //     selectedCalendars: [TestData.selectedCalendars.google],
+  //   });
+
+  const pro30MinEventType = await prisma.eventType.findFirst({
+    where: {
+      slug: "30min",
+      hosts: {
+        some: {
+          user: {
+            email: "pro@example.com",
+          },
+        },
+      },
+    },
+  });
+
+  const mockBookingData = getMockRequestDataForBooking({
+    data: {
+      eventTypeId: pro30MinEventType?.id,
+      responses: {
+        email: booker.email,
+        name: booker.name,
+        location: { optionValue: "", value: "integrations:daily" },
+      },
+    },
+  });
+
+  const { req, res } = createMocks<CustomNextApiRequest, CustomNextApiResponse>({
+    method: "POST",
+    body: mockBookingData,
+    prisma,
+  });
+  //   console.log("🚀 ~ file: handleNewBooking.test.ts:56 ~ test ~ prisma:", prisma);
+
+  //   const { req, res } = createMockNextJsRequest({
+  //     method: "POST",
+  //     body: mockBookingData,
+  //     prisma,
+  //   });
+
+  withPrisma(async (req: NextApiRequest, res: NextApiResponse) => {
+    // Your code here
+    handleNewBooking(req);
```

Comment:
Just experimenting with this. The idea was to add prisma to the request if it wasn't initialized.